src/billing/components/PaymentInfo/PaymentPanelBody.tsx

```diff
@@ -1,51 +1,72 @@
 // Libraries
-import React, {FC, useState} from 'react'
+import React, {FC, useCallback, useEffect, useState} from 'react'
 
 // Components
 import {Panel, ComponentSize} from '@influxdata/clockface'
 import PaymentDisplay from './PaymentDisplay'
 import PaymentForm from './PaymentForm'
 
 // Types
-import {CreditCardParams, ZuoraResponse} from 'src/types/billing'
+import {CreditCardParams} from 'src/types/billing'
+import {RemoteDataState} from 'src/types'
+
+import {
+  getBillingCreditCardParams,
+  putBillingPaymentMethodId,
+} from 'src/billing/api'
+import {getErrorMessage} from 'src/utils/api'
 
 interface Props {
   isEditing: boolean
   onCancel: () => void
 }
 
+const EMPTY_CREDIT_CARD_PARAMS: CreditCardParams = {
+  id: '',
+  tenantId: '',
+  key: '',
+  signature: '',
+  token: '',
+  style: '',
+  submitEnabled: 'false',
+  url: '',
+  status: RemoteDataState.NotStarted,
+}
+
 const PaymentPanelBody: FC<Props> = ({isEditing, onCancel}) => {
   const [errorMessage, setErrorMessage] = useState('')
+  const [creditCardParams, setCreditCardParams] = useState(
+    EMPTY_CREDIT_CARD_PARAMS
+  )
 
-  const onSubmit = async (response: ZuoraResponse): Promise<void> => {
-    const error = 'Could not add card, please try again.'
-    if (response?.success) {
-      try {
-        const url = 'privateAPI/billing/payment_method'
-        const data = await fetch(url, {
-          method: 'PUT',
-          body: JSON.stringify({
-            paymentMethodId: response.refId,
-          }),
-        })
-        onCancel()
-        setErrorMessage('')
-        // TODO(ariel): refetch the updated payment methods elsewhere once this has resolved
-        console.log(data) // eslint-disable-line no-console
-      } catch {
-        setErrorMessage(error)
-      }
+  const onSubmit = async (paymentMethodId: string): Promise<void> => {
+    const response = await putBillingPaymentMethodId(paymentMethodId)
+    if (response.status !== 200) {
+      setErrorMessage(getErrorMessage(response))
     } else {
-      setErrorMessage(error)
+      onCancel()
+      setErrorMessage('')
```

Comment:
We should probably add some sort of notification saying that there was an issue if updating the payment method fails

Reply:
This is basically clearing any messages previously displayed. The error is being displayed on line 45.

Reply:
Aw, i see what you're saying